pygmt/src/surface.py

```diff
@@ -99,11 +99,4 @@ def surface(x=None, y=None, z=None, data=None, **kwargs):
                 arg_str = " ".join([infile, build_arg_string(kwargs)])
                 lib.call_module(module="surface", args=arg_str)
 
-        if outfile == tmpfile.name:  # if user did not set outfile, return DataArray
-            with xr.open_dataarray(outfile) as dataarray:
-                result = dataarray.load()
-                _ = result.gmt  # load GMTDataArray accessor information
-        elif outfile != tmpfile.name:  # if user sets an outfile, return None
-            result = None
-
-    return result
+        return process_output_grid(outfile, tmpfile.name)
```

Comment:
Just a note that we should probably standardize on using `outgrid` instead of `outfile` here for `pygmt.surface`. Will need to go through some deprecation cycles :slightly_smiling_face:

Reply:
Completely agree :smile: Want to open an issue for this?

Reply:
Ok, done at #1441.